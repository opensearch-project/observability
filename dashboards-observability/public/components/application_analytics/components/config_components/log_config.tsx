/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiAccordion,
  EuiText,
  EuiSpacer,
  EuiButton,
  EuiFormRow,
  EuiFlexItem,
  EuiBadge,
  EuiOverlayMask,
  EuiCallOut,
  EuiFlexGroup,
} from '@elastic/eui';
import DSLService from 'public/services/requests/dsl';
import React, { useState } from 'react';
import {
  parseGetSuggestions,
  onItemSelect,
} from '../../../../../public/components/common/search/autocomplete_logic';
import { uiSettingsService } from '../../../../../common/utils';
import { Autocomplete } from '../../../common/search/autocomplete';
import { AppAnalyticsComponentDeps } from '../../home';
import { getClearModal } from '../../helpers/modal_containers';
import '../../app_analytics.scss';

interface LogConfigProps extends AppAnalyticsComponentDeps {
  dslService: DSLService;
  setIsFlyoutVisible: any;
  editMode: boolean;
}

export const LogConfig = (props: LogConfigProps) => {
  const { dslService, query, setQueryWithStorage, setIsFlyoutVisible, editMode } = props;
  const [logOpen, setLogOpen] = useState(false);
  const [isModalVisible, setIsModalVisible] = useState(false);
  const [modalLayout, setModalLayout] = useState(<EuiOverlayMask />);
  const [tempQuery, setTempQuery] = useState<string>('');

  const handleQueryChange = async (newQuery: string) => {
    setTempQuery(newQuery);
    setQueryWithStorage(newQuery);
  };

  const showFlyout = () => {
    setIsFlyoutVisible(true);
  };

  const onCancel = () => {
    setIsModalVisible(false);
  };

  const closeModal = () => {
    setIsModalVisible(false);
  };

  const showModal = () => {
    setIsModalVisible(true);
  };

  const onConfirm = () => {
    handleQueryChange('');
    closeModal();
  };

  const clearAllModal = () => {
    setModalLayout(
      getClearModal(
        onCancel,
        onConfirm,
        'Clear log source?',
        'This will clear all information in log source configuration.',
        'Clear'
      )
    );
    showModal();
  };

  const allowedCommands = [
    { label: 'dedup' },
    { label: 'eval' },
    { label: 'fields' },
    { label: 'parse' },
    { label: 'rename' },
    { label: 'sort' },
    { label: 'where' },
  ];

  return (
    <div>
      <EuiAccordion
        id="logSource"
        data-test-subj="logSourceAccordion"
        buttonContent={
          <>
            <EuiText size="s">
              <h3>Log source</h3>
            </EuiText>
            <EuiSpacer size="s" />
            <EuiText size="s" color="subdued">
              Configure your application base query
            </EuiText>
          </>
        }
        extraAction={
          <EuiButton
            size="s"
            data-test-subj="clearLogSourceButton"
            disabled={!logOpen || !query.length || editMode}
            onClick={clearAllModal}
          >
            Clear
          </EuiButton>
        }
        onToggle={(isOpen) => {
          setLogOpen(isOpen);
        }}
        paddingSize="l"
      >
        <EuiFlexGroup direction="column" gutterSize="s">
          <EuiFlexItem>
            <EuiCallOut id="baseQueryCallout" iconType={'iInCircle'} size="s">
              You can&apos;t change the base query after the application is created.
            </EuiCallOut>
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiFormRow
              label="Base Query"
              helpText="The default logs view in the application will be filtered by this query."
            >
              <EuiFlexItem grow={false} key="query-bar" className="query-area">
                <Autocomplete
                  key={'autocomplete-bar'}
                  query={query}
                  tempQuery={tempQuery}
                  baseQuery=""
                  handleQueryChange={handleQueryChange}
                  handleQuerySearch={() => {}}
                  dslService={dslService}
                  getSuggestions={parseGetSuggestions}
                  onItemSelect={onItemSelect}
                  isDisabled={editMode}
                  tabId={'application-analytics-tab'}
                  possibleCommands={allowedCommands}
                />
                <EuiBadge
                  className={`ppl-link ${
                    uiSettingsService.get('theme:darkMode') ? 'ppl-link-dark' : 'ppl-link-light'
                  }`}
                  color="hollow"
                  onClick={showFlyout}
                  onClickAriaLabel={'pplLinkShowFlyout'}
                >
                  PPL
                </EuiBadge>
              </EuiFlexItem>
            </EuiFormRow>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiAccordion>
      {isModalVisible && modalLayout}
    </div>
  );
};
