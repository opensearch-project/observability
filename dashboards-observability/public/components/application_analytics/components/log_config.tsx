/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiAccordion, EuiText, EuiSpacer, EuiButton, EuiFormRow, EuiFlexItem, EuiBadge } from "@elastic/eui";
import { uiSettingsService } from "../../../../common/utils";
import { Autocomplete } from "../../../components/common/search/autocomplete";
import DSLService from "public/services/requests/dsl";
import React, { useState } from "react";
import { AppAnalyticsComponentDeps } from "../home";

interface LogConfigProps extends AppAnalyticsComponentDeps {
  dslService: DSLService;
  setIsFlyoutVisible: any;
}

export const LogConfig = (props: LogConfigProps) => {
  const { dslService, query, setQuery, setIsFlyoutVisible } = props;
  const [logOpen, setLogOpen] = useState(false);
  const [tempQuery, setTempQuery] = useState<string>('');

  const handleQueryChange = async (query: string) => setQuery(query);

  const showFlyout = () => {
    setIsFlyoutVisible(true);
  };

  return (
    <EuiAccordion
      id="logSource"
      buttonContent={
        <>
        <EuiText size="s">
        <h3>Log Source</h3>
        </EuiText>
        <EuiSpacer size="s" />
        <EuiText size="s" color="subdued">
        Configure your application base query
        </EuiText>
        </>
      }
      extraAction={<EuiButton size="s" disabled={!logOpen} onClick={() => { handleQueryChange('') }}>Clear all</EuiButton>}
      onToggle={(isOpen) => {setLogOpen(isOpen)}}
      paddingSize="l"
    >
      <EuiFormRow
      label="PPL Base Query"
      helpText="The default logs view in the application will be filtered by this query."
      >
        <EuiFlexItem grow={false} key="query-bar" className="query-area">
          <Autocomplete
            key={'autocomplete-bar'}
            query={query}
            tempQuery={tempQuery}
            handleQueryChange={handleQueryChange}
            handleQuerySearch={() => {}}
            dslService={dslService}
          />
          <EuiBadge 
            className={`ppl-link ${uiSettingsService.get('theme:darkMode') ? "ppl-link-dark" : "ppl-link-light"}`}
            color="hollow"
            onClick={() => showFlyout()}
            onClickAriaLabel={"pplLinkShowFlyout"}
          >
            PPL
          </EuiBadge>
        </EuiFlexItem>
      </EuiFormRow>
    </EuiAccordion>
  );
}