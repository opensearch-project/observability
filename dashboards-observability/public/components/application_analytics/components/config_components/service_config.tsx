/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
/* eslint-disable react-hooks/exhaustive-deps */

import {
  EuiAccordion,
  EuiBadge,
  EuiButton,
  EuiComboBox,
  EuiFormRow,
  EuiOverlayMask,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';
import DSLService from 'public/services/requests/dsl';
import React, { useState } from 'react';
import { useEffect } from 'react';
import { FilterType } from '../../../trace_analytics/components/common/filters/filters';
import { ServiceObject } from '../../../trace_analytics/components/common/plots/service_map';
import { ServiceMap } from '../../../trace_analytics/components/services';
import { handleServiceMapRequest } from '../../../trace_analytics/requests/services_request_handler';
import { AppAnalyticsComponentDeps } from '../../home';
import { OptionType } from '../../../../../common/types/app_analytics';
import { getClearModal } from '../../helpers/modal_containers';

interface ServiceConfigProps extends AppAnalyticsComponentDeps {
  dslService: DSLService;
  selectedServices: OptionType[];
  setSelectedServices: (services: OptionType[]) => void;
}

export const ServiceConfig = (props: ServiceConfigProps) => {
  const {
    dslService,
    filters,
    setFiltersWithStorage,
    http,
    selectedServices,
    setSelectedServices,
  } = props;
  const [servicesOpen, setServicesOpen] = useState(false);
  const [serviceMap, setServiceMap] = useState<ServiceObject>({});
  const [serviceMapIdSelected, setServiceMapIdSelected] = useState<
    'latency' | 'error_rate' | 'throughput'
  >('latency');
  const [isModalVisible, setIsModalVisible] = useState(false);
  const [modalLayout, setModalLayout] = useState(<EuiOverlayMask />);

  useEffect(() => {
    handleServiceMapRequest(http, dslService, setServiceMap);
  }, []);

  useEffect(() => {
    const serviceOptions = filters
      .filter((f) => f.field === 'serviceName')
      .map((f) => {
        return { label: f.value };
      });
    const noDups = serviceOptions.filter((s, index) => {
      return serviceOptions.findIndex((ser) => ser.label === s.label) === index;
    });
    setSelectedServices(noDups);
  }, [filters]);

  const addFilter = (filter: FilterType) => {
    for (const addedFilter of filters) {
      if (
        addedFilter.field === filter.field &&
        addedFilter.operator === filter.operator &&
        addedFilter.value === filter.value
      ) {
        return;
      }
    }
    const newFilters = [...filters, filter];
    setFiltersWithStorage(newFilters);
  };

  const onServiceChange = (newServices: any) => {
    const serviceFilters = newServices.map((option: OptionType) => {
      return {
        field: 'serviceName',
        operator: 'is',
        value: option.label,
        inverted: false,
        disabled: false,
      };
    });
    const nonServiceFilters = filters.filter((f) => f.field !== 'serviceName');
    setFiltersWithStorage([...nonServiceFilters, ...serviceFilters]);
  };

  const clearServices = () => {
    const withoutServices = filters.filter((f) => f.field !== 'serviceName');
    setFiltersWithStorage(withoutServices);
  };

  const services = Object.keys(serviceMap).map((service) => {
    return { label: decodeURI(service) };
  });

  const onCancel = () => {
    setIsModalVisible(false);
  };

  const closeModal = () => {
    setIsModalVisible(false);
  };

  const showModal = () => {
    setIsModalVisible(true);
  };

  const onConfirm = () => {
    clearServices();
    closeModal();
  };

  const clearAllModal = () => {
    setModalLayout(
      getClearModal(
        onCancel,
        onConfirm,
        'Clear services & entities?',
        'This will clear all information in services & entities configuration.',
        'Clear all'
      )
    );
    showModal();
  };

  return (
    <div>
      <EuiAccordion
        id="servicesEntities"
        data-test-subj="servicesEntitiesAccordion"
        buttonContent={
          <>
            <EuiText size="s">
              <h3>
                Services & entities{' '}
                <EuiBadge data-test-subj="servicesEntitiesCountBadge">
                  {selectedServices.length}
                </EuiBadge>
              </h3>
            </EuiText>
            <EuiSpacer size="s" />
            <EuiText size="s" color="subdued">
              Select services & entities to include in this application
            </EuiText>
          </>
        }
        extraAction={
          <EuiButton
            size="s"
            data-test-subj="clearServicesEntitiesButton"
            disabled={!servicesOpen || !selectedServices.length}
            onClick={clearAllModal}
          >
            Clear all
          </EuiButton>
        }
        onToggle={(isOpen) => {
          setServicesOpen(isOpen);
        }}
        paddingSize="l"
      >
        <EuiFormRow label="Services & entities">
          <EuiComboBox
            aria-label="Select services and entities"
            placeholder="Select services and entities"
            options={services}
            selectedOptions={selectedServices}
            onChange={onServiceChange}
            isClearable={false}
            data-test-subj="servicesEntitiesComboBox"
          />
        </EuiFormRow>
        <EuiSpacer />
        <ServiceMap
          serviceMap={serviceMap}
          idSelected={serviceMapIdSelected}
          setIdSelected={setServiceMapIdSelected}
          addFilter={addFilter}
          page="appCreate"
        />
      </EuiAccordion>
      {isModalVisible && modalLayout}
    </div>
  );
};
