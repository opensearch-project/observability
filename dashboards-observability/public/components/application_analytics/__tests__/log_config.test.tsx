/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { LogConfig } from '../components/config_components/log_config';
import { coreStartMock } from '../../../../test/__mocks__/coreMocks';
import DSLService from 'public/services/requests/dsl';

describe('Log Config component', () => {
  configure({ adapter: new Adapter() });

  it('renders empty log config', () => {
    const core = coreStartMock;
    const setQuery = jest.fn();
    const setFilters = jest.fn();
    const setStartTime = jest.fn();
    const setEndTime = jest.fn();
    const setIsFlyoutVisible = jest.fn();
    const dslService = {
      http: jest.fn(),
      fetch: jest.fn(),
      fetchIndices: jest.fn(),
      fetchFields: jest.fn()
    } as unknown as DSLService;
    const wrapper = mount(
      <LogConfig
        http={core.http}
        chrome={core.chrome}
        parentBreadcrumb={{ text: 'test', href: 'test#/' }}
        query=""
        setQuery={setQuery}
        filters={[]}
        setFilters={setFilters}
        startTime="now-5m"
        setStartTime={setStartTime}
        endTime="now"
        setEndTime={setEndTime}
        indicesExist={true} 
        dslService={dslService}
        setIsFlyoutVisible={setIsFlyoutVisible}
        />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with query', () => {
    const core = coreStartMock;
    const setQuery = jest.fn();
    const setFilters = jest.fn();
    const setStartTime = jest.fn();
    const setEndTime = jest.fn();
    const setIsFlyoutVisible = jest.fn();
    const dslService = {
      http: jest.fn(),
      fetch: jest.fn(),
      fetchIndices: jest.fn(),
      fetchFields: jest.fn()
    } as unknown as DSLService;
    const wrapper = mount(
      <LogConfig
        http={core.http}
        chrome={core.chrome}
        parentBreadcrumb={{ text: 'test', href: 'test#/' }}
        query="source = openserach_dashboard_sample_logs"
        setQuery={setQuery}
        filters={[]}
        setFilters={setFilters}
        startTime="now-5m"
        setStartTime={setStartTime}
        endTime="now"
        setEndTime={setEndTime}
        indicesExist={true} 
        dslService={dslService}
        setIsFlyoutVisible={setIsFlyoutVisible}
        />
    );

    expect(wrapper).toMatchSnapshot();
  });
});