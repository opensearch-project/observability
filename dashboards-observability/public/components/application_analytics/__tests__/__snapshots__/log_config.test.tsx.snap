// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log Config component renders empty log config 1`] = `
<LogConfig
  chrome={
    Object {
      "getIsNavDrawerLocked$": [MockFunction],
      "setBreadcrumbs": [MockFunction],
    }
  }
  description=""
  dslService={
    Object {
      "fetch": [MockFunction],
      "fetchFields": [MockFunction],
      "fetchIndices": [MockFunction],
      "http": [MockFunction],
    }
  }
  endTime="now"
  filters={Array []}
  http={[MockFunction]}
  indicesExist={true}
  name=""
  parentBreadcrumb={
    Object {
      "href": "test#/",
      "text": "test",
    }
  }
  query=""
  setDescriptionWithStorage={[MockFunction]}
  setEndTime={[MockFunction]}
  setFilters={[MockFunction]}
  setFiltersWithStorage={[MockFunction]}
  setIsFlyoutVisible={[MockFunction]}
  setNameWithStorage={[MockFunction]}
  setQuery={[MockFunction]}
  setQueryWithStorage={[MockFunction]}
  setStartTime={[MockFunction]}
  startTime="now-5m"
>
  <div>
    <EuiAccordion
      arrowDisplay="left"
      buttonContent={
        <React.Fragment>
          <EuiText
            size="s"
          >
            <h3>
              Log Source
            </h3>
          </EuiText>
          <EuiSpacer
            size="s"
          />
          <EuiText
            color="subdued"
            size="s"
          >
            Configure your application base query
          </EuiText>
        </React.Fragment>
      }
      extraAction={
        <EuiButton
          disabled={true}
          onClick={[Function]}
          size="s"
        >
          Clear
        </EuiButton>
      }
      id="logSource"
      initialIsOpen={false}
      isLoading={false}
      isLoadingMessage={false}
      onToggle={[Function]}
      paddingSize="l"
    >
      <div
        className="euiAccordion"
        onToggle={[Function]}
      >
        <div
          className="euiAccordion__triggerWrapper"
        >
          <button
            aria-controls="logSource"
            aria-expanded={false}
            className="euiAccordion__button"
            id="random_html_id"
            onClick={[Function]}
            type="button"
          >
            <span
              className="euiAccordion__iconWrapper"
            >
              <EuiIcon
                className="euiAccordion__icon"
                size="m"
                type="arrowRight"
              >
                <EuiIconEmpty
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
                  focusable="false"
                  role="img"
                  style={null}
                >
                  <svg
                    aria-hidden={true}
                    className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
                    focusable="false"
                    height={16}
                    role="img"
                    style={null}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </EuiIconEmpty>
              </EuiIcon>
            </span>
            <span
              className="euiIEFlexWrapFix"
            >
              <EuiText
                size="s"
              >
                <div
                  className="euiText euiText--small"
                >
                  <h3>
                    Log Source
                  </h3>
                </div>
              </EuiText>
              <EuiSpacer
                size="s"
              >
                <div
                  className="euiSpacer euiSpacer--s"
                />
              </EuiSpacer>
              <EuiText
                color="subdued"
                size="s"
              >
                <div
                  className="euiText euiText--small"
                >
                  <EuiTextColor
                    color="subdued"
                    component="div"
                  >
                    <div
                      className="euiTextColor euiTextColor--subdued"
                    >
                      Configure your application base query
                    </div>
                  </EuiTextColor>
                </div>
              </EuiText>
            </span>
          </button>
          <div
            className="euiAccordion__optionalAction"
          >
            <EuiButton
              disabled={true}
              onClick={[Function]}
              size="s"
            >
              <EuiButtonDisplay
                disabled={true}
                element="button"
                isDisabled={true}
                onClick={[Function]}
                size="s"
                type="button"
              >
                <button
                  className="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
                  disabled={true}
                  onClick={[Function]}
                  type="button"
                >
                  <EuiButtonContent
                    className="euiButton__content"
                    iconSide="left"
                    textProps={
                      Object {
                        "className": "euiButton__text",
                      }
                    }
                  >
                    <span
                      className="euiButtonContent euiButton__content"
                    >
                      <span
                        className="euiButton__text"
                      >
                        Clear
                      </span>
                    </span>
                  </EuiButtonContent>
                </button>
              </EuiButtonDisplay>
            </EuiButton>
          </div>
        </div>
        <div
          className="euiAccordion__childWrapper"
          id="logSource"
        >
          <EuiResizeObserver
            onResize={[Function]}
          >
            <div>
              <div
                className="euiAccordion__padding--l"
              >
                <EuiFormRow
                  describedByIds={Array []}
                  display="row"
                  fullWidth={false}
                  hasChildLabel={true}
                  hasEmptyLabelSpace={false}
                  helpText="The default logs view in the application will be filtered by this query."
                  label="PPL Base Query"
                  labelType="label"
                >
                  <div
                    className="euiFormRow"
                    id="random_html_id-row"
                  >
                    <div
                      className="euiFormRow__labelWrapper"
                    >
                      <EuiFormLabel
                        className="euiFormRow__label"
                        htmlFor="random_html_id"
                        isFocused={false}
                        type="label"
                      >
                        <label
                          className="euiFormLabel euiFormRow__label"
                          htmlFor="random_html_id"
                        >
                          PPL Base Query
                        </label>
                      </EuiFormLabel>
                    </div>
                    <div
                      className="euiFormRow__fieldWrapper"
                    >
                      <EuiFlexItem
                        aria-describedby="random_html_id-help"
                        className="query-area"
                        grow={false}
                        id="random_html_id"
                        key="query-bar"
                        onBlur={[Function]}
                        onFocus={[Function]}
                      >
                        <div
                          aria-describedby="random_html_id-help"
                          className="euiFlexItem euiFlexItem--flexGrowZero query-area"
                          id="random_html_id"
                          onBlur={[Function]}
                          onFocus={[Function]}
                        >
                          <Autocomplete
                            dslService={
                              Object {
                                "fetch": [MockFunction],
                                "fetchFields": [MockFunction],
                                "fetchIndices": [MockFunction],
                                "http": [MockFunction],
                              }
                            }
                            getSuggestions={[Function]}
                            handleQueryChange={[Function]}
                            handleQuerySearch={[Function]}
                            key="autocomplete-bar"
                            onItemSelect={[Function]}
                            query=""
                            tempQuery=""
                          >
                            <div
                              aria-expanded={false}
                              aria-haspopup="listbox"
                              aria-labelledby="autocomplete-0-label"
                              className="aa-Autocomplete"
                              id="autocomplete-root"
                              role="combobox"
                            >
                              <EuiTextArea
                                aria-autocomplete="both"
                                aria-labelledby="autocomplete-0-label"
                                autoCapitalize="off"
                                autoComplete="off"
                                autoCorrect="off"
                                autoFocus={false}
                                data-test-subj="searchAutocompleteTextArea"
                                enterKeyHint="search"
                                id="autocomplete-textarea"
                                maxLength={512}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                placeholder="Enter PPL query"
                                spellCheck="false"
                                type="search"
                                value=""
                              >
                                <EuiValidatableControl>
                                  <textarea
                                    aria-autocomplete="both"
                                    aria-labelledby="autocomplete-0-label"
                                    autoCapitalize="off"
                                    autoComplete="off"
                                    autoCorrect="off"
                                    autoFocus={false}
                                    className="euiTextArea euiTextArea--resizeVertical"
                                    data-test-subj="searchAutocompleteTextArea"
                                    enterKeyHint="search"
                                    id="autocomplete-textarea"
                                    maxLength={512}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="Enter PPL query"
                                    rows={6}
                                    spellCheck="false"
                                    type="search"
                                    value=""
                                  />
                                </EuiValidatableControl>
                              </EuiTextArea>
                            </div>
                          </Autocomplete>
                          <EuiBadge
                            className="ppl-link ppl-link-light"
                            color="hollow"
                            onClick={[Function]}
                            onClickAriaLabel="pplLinkShowFlyout"
                          >
                            <EuiInnerText>
                              <button
                                aria-label="pplLinkShowFlyout"
                                className="euiBadge euiBadge-isClickable euiBadge--hollow euiBadge--iconLeft ppl-link ppl-link-light"
                                disabled={false}
                                onClick={[Function]}
                                title="PPL"
                              >
                                <span
                                  className="euiBadge__content"
                                >
                                  <span
                                    className="euiBadge__text"
                                  >
                                    PPL
                                  </span>
                                </span>
                              </button>
                            </EuiInnerText>
                          </EuiBadge>
                        </div>
                      </EuiFlexItem>
                      <EuiFormHelpText
                        className="euiFormRow__text"
                        id="random_html_id-help"
                      >
                        <div
                          className="euiFormHelpText euiFormRow__text"
                          id="random_html_id-help"
                        >
                          The default logs view in the application will be filtered by this query.
                        </div>
                      </EuiFormHelpText>
                    </div>
                  </div>
                </EuiFormRow>
              </div>
            </div>
          </EuiResizeObserver>
        </div>
      </div>
    </EuiAccordion>
  </div>
</LogConfig>
`;

exports[`Log Config component renders with query 1`] = `
<LogConfig
  chrome={
    Object {
      "getIsNavDrawerLocked$": [MockFunction],
      "setBreadcrumbs": [MockFunction],
    }
  }
  description=""
  dslService={
    Object {
      "fetch": [MockFunction],
      "fetchFields": [MockFunction],
      "fetchIndices": [MockFunction],
      "http": [MockFunction],
    }
  }
  endTime="now"
  filters={Array []}
  http={[MockFunction]}
  indicesExist={true}
  name=""
  parentBreadcrumb={
    Object {
      "href": "test#/",
      "text": "test",
    }
  }
  query="source = openserach_dashboard_sample_logs"
  setDescriptionWithStorage={[MockFunction]}
  setEndTime={[MockFunction]}
  setFilters={[MockFunction]}
  setFiltersWithStorage={[MockFunction]}
  setIsFlyoutVisible={[MockFunction]}
  setNameWithStorage={[MockFunction]}
  setQuery={[MockFunction]}
  setQueryWithStorage={[MockFunction]}
  setStartTime={[MockFunction]}
  startTime="now-5m"
>
  <div>
    <EuiAccordion
      arrowDisplay="left"
      buttonContent={
        <React.Fragment>
          <EuiText
            size="s"
          >
            <h3>
              Log Source
            </h3>
          </EuiText>
          <EuiSpacer
            size="s"
          />
          <EuiText
            color="subdued"
            size="s"
          >
            Configure your application base query
          </EuiText>
        </React.Fragment>
      }
      extraAction={
        <EuiButton
          disabled={true}
          onClick={[Function]}
          size="s"
        >
          Clear
        </EuiButton>
      }
      id="logSource"
      initialIsOpen={false}
      isLoading={false}
      isLoadingMessage={false}
      onToggle={[Function]}
      paddingSize="l"
    >
      <div
        className="euiAccordion"
        onToggle={[Function]}
      >
        <div
          className="euiAccordion__triggerWrapper"
        >
          <button
            aria-controls="logSource"
            aria-expanded={false}
            className="euiAccordion__button"
            id="random_html_id"
            onClick={[Function]}
            type="button"
          >
            <span
              className="euiAccordion__iconWrapper"
            >
              <EuiIcon
                className="euiAccordion__icon"
                size="m"
                type="arrowRight"
              >
                <EuiIconArrowRight
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiAccordion__icon"
                  focusable="false"
                  role="img"
                  style={null}
                >
                  <svg
                    aria-hidden={true}
                    className="euiIcon euiIcon--medium euiAccordion__icon"
                    focusable="false"
                    height={16}
                    role="img"
                    style={null}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.157 13.069l4.611-4.685a.546.546 0 000-.768L5.158 2.93a.552.552 0 010-.771.53.53 0 01.759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 01-.76 0 .552.552 0 010-.771z"
                      fillRule="nonzero"
                    />
                  </svg>
                </EuiIconArrowRight>
              </EuiIcon>
            </span>
            <span
              className="euiIEFlexWrapFix"
            >
              <EuiText
                size="s"
              >
                <div
                  className="euiText euiText--small"
                >
                  <h3>
                    Log Source
                  </h3>
                </div>
              </EuiText>
              <EuiSpacer
                size="s"
              >
                <div
                  className="euiSpacer euiSpacer--s"
                />
              </EuiSpacer>
              <EuiText
                color="subdued"
                size="s"
              >
                <div
                  className="euiText euiText--small"
                >
                  <EuiTextColor
                    color="subdued"
                    component="div"
                  >
                    <div
                      className="euiTextColor euiTextColor--subdued"
                    >
                      Configure your application base query
                    </div>
                  </EuiTextColor>
                </div>
              </EuiText>
            </span>
          </button>
          <div
            className="euiAccordion__optionalAction"
          >
            <EuiButton
              disabled={true}
              onClick={[Function]}
              size="s"
            >
              <EuiButtonDisplay
                disabled={true}
                element="button"
                isDisabled={true}
                onClick={[Function]}
                size="s"
                type="button"
              >
                <button
                  className="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
                  disabled={true}
                  onClick={[Function]}
                  type="button"
                >
                  <EuiButtonContent
                    className="euiButton__content"
                    iconSide="left"
                    textProps={
                      Object {
                        "className": "euiButton__text",
                      }
                    }
                  >
                    <span
                      className="euiButtonContent euiButton__content"
                    >
                      <span
                        className="euiButton__text"
                      >
                        Clear
                      </span>
                    </span>
                  </EuiButtonContent>
                </button>
              </EuiButtonDisplay>
            </EuiButton>
          </div>
        </div>
        <div
          className="euiAccordion__childWrapper"
          id="logSource"
        >
          <EuiResizeObserver
            onResize={[Function]}
          >
            <div>
              <div
                className="euiAccordion__padding--l"
              >
                <EuiFormRow
                  describedByIds={Array []}
                  display="row"
                  fullWidth={false}
                  hasChildLabel={true}
                  hasEmptyLabelSpace={false}
                  helpText="The default logs view in the application will be filtered by this query."
                  label="PPL Base Query"
                  labelType="label"
                >
                  <div
                    className="euiFormRow"
                    id="random_html_id-row"
                  >
                    <div
                      className="euiFormRow__labelWrapper"
                    >
                      <EuiFormLabel
                        className="euiFormRow__label"
                        htmlFor="random_html_id"
                        isFocused={false}
                        type="label"
                      >
                        <label
                          className="euiFormLabel euiFormRow__label"
                          htmlFor="random_html_id"
                        >
                          PPL Base Query
                        </label>
                      </EuiFormLabel>
                    </div>
                    <div
                      className="euiFormRow__fieldWrapper"
                    >
                      <EuiFlexItem
                        aria-describedby="random_html_id-help"
                        className="query-area"
                        grow={false}
                        id="random_html_id"
                        key="query-bar"
                        onBlur={[Function]}
                        onFocus={[Function]}
                      >
                        <div
                          aria-describedby="random_html_id-help"
                          className="euiFlexItem euiFlexItem--flexGrowZero query-area"
                          id="random_html_id"
                          onBlur={[Function]}
                          onFocus={[Function]}
                        >
                          <Autocomplete
                            dslService={
                              Object {
                                "fetch": [MockFunction],
                                "fetchFields": [MockFunction],
                                "fetchIndices": [MockFunction],
                                "http": [MockFunction],
                              }
                            }
                            getSuggestions={[Function]}
                            handleQueryChange={[Function]}
                            handleQuerySearch={[Function]}
                            key="autocomplete-bar"
                            onItemSelect={[Function]}
                            query="source = openserach_dashboard_sample_logs"
                            tempQuery=""
                          >
                            <div
                              aria-expanded={false}
                              aria-haspopup="listbox"
                              aria-labelledby="autocomplete-1-label"
                              className="aa-Autocomplete"
                              id="autocomplete-root"
                              role="combobox"
                            >
                              <EuiTextArea
                                aria-autocomplete="both"
                                aria-labelledby="autocomplete-1-label"
                                autoCapitalize="off"
                                autoComplete="off"
                                autoCorrect="off"
                                autoFocus={false}
                                data-test-subj="searchAutocompleteTextArea"
                                enterKeyHint="search"
                                id="autocomplete-textarea"
                                maxLength={512}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                placeholder="Enter PPL query"
                                spellCheck="false"
                                type="search"
                                value="source = openserach_dashboard_sample_logs"
                              >
                                <EuiValidatableControl>
                                  <textarea
                                    aria-autocomplete="both"
                                    aria-labelledby="autocomplete-1-label"
                                    autoCapitalize="off"
                                    autoComplete="off"
                                    autoCorrect="off"
                                    autoFocus={false}
                                    className="euiTextArea euiTextArea--resizeVertical"
                                    data-test-subj="searchAutocompleteTextArea"
                                    enterKeyHint="search"
                                    id="autocomplete-textarea"
                                    maxLength={512}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="Enter PPL query"
                                    rows={6}
                                    spellCheck="false"
                                    type="search"
                                    value="source = openserach_dashboard_sample_logs"
                                  />
                                </EuiValidatableControl>
                              </EuiTextArea>
                            </div>
                          </Autocomplete>
                          <EuiBadge
                            className="ppl-link ppl-link-light"
                            color="hollow"
                            onClick={[Function]}
                            onClickAriaLabel="pplLinkShowFlyout"
                          >
                            <EuiInnerText>
                              <button
                                aria-label="pplLinkShowFlyout"
                                className="euiBadge euiBadge-isClickable euiBadge--hollow euiBadge--iconLeft ppl-link ppl-link-light"
                                disabled={false}
                                onClick={[Function]}
                                title="PPL"
                              >
                                <span
                                  className="euiBadge__content"
                                >
                                  <span
                                    className="euiBadge__text"
                                  >
                                    PPL
                                  </span>
                                </span>
                              </button>
                            </EuiInnerText>
                          </EuiBadge>
                        </div>
                      </EuiFlexItem>
                      <EuiFormHelpText
                        className="euiFormRow__text"
                        id="random_html_id-help"
                      >
                        <div
                          className="euiFormHelpText euiFormRow__text"
                          id="random_html_id-help"
                        >
                          The default logs view in the application will be filtered by this query.
                        </div>
                      </EuiFormHelpText>
                    </div>
                  </div>
                </EuiFormRow>
              </div>
            </div>
          </EuiResizeObserver>
        </div>
      </div>
    </EuiAccordion>
  </div>
</LogConfig>
`;
